### DISCUSSION TITLE
How to tweet image ?

### DISCUSSION THREAD

**Discussion by d4nm0** [tweepy/tweepy#1954] [Questions] [answered]
Hello, how to tweet image with client create_tweet ?

response = client.create_tweet(
        text="text test"
    )

**Comment by 4RCAN3** [comment]
Media upload endpoint is not yet available on v2. You can add media ID's for the media that has been uploaded using v1.1. Reference: https://developer.twitter.com/en/docs/twitter-api/tweets/manage-tweets/integrate#media

**Comment by Trevordiex** [comment]
v1.1 has an API for uploading image with tweet.
`api.update_status_with_media(status,filename,*,file,**kwargs)`

unfortunately, the twitter endpoint for this method is deprecated. For now the only way to upload a media is to upload it seperately via the `media_upload`  endpoint and include the id with with the status.

`media_id = api.upload_media(filename,file=<file object>, **kwargs).media_id`
`api.update_status(status,media_ids=[media_id,])`

you can include up to four images in a single status

**Comment by Harmon758** [comment, marked as answer]
You can use [`API.media_upload`](https://tweepy.readthedocs.io/en/v4.10.1/api.html#tweepy.API.media_upload) (or any of the other `API` media upload methods) to upload media.
You can then use the Media IDs to associate them with a Tweet when creating the Tweet.

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

@Trevordiex `API.upload_media` is not an existing method.
